### PR TITLE
[T4A3][W11-B4]Ong Heng Le

### PR DESCRIPTION
**Pull request by initialshl**

### CODE REVIEW

**Comment by initialshl** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: can you see that with this equals test, we can match `Address` with `Phone`. think why?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -6,15 +6,12 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: what about the data members? how are you setting them? i don't see a call to `super()` here

Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: also think where would be the right place to have the `equals()` method and the validation.

please recollect the discussions during tutorial to help understand inheritance better.

**Comment by okkhoy** [comment]
@initialshl 
Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**